Realm/RLMQueryUtil.mm

```diff
@@ -538,6 +517,43 @@ void add_constraint_to_query(realm::Query &query, RLMPropertyType type,
     }
 }
 
+void do_add_constraint_to_query(realm::Query&, RLMPropertyType, NSPredicateOperatorType,
+                                NSComparisonPredicateOptions, const std::vector<NSUInteger>&,
+                                id, realm::null)
+{
+    // This probably isn't actually reachable since there's earlier checks for
+    // it, but they're done before null is un-type-erased
+    @throw RLMPredicateException(@"Invalid predicate expressions",
```

Comment:
This definitely isn't reachable due to the logic in `update_query_with_predicate`. I'm guessing this overload is necessary in order for `add_constraint_to_query` to compile?